### PR TITLE
Delete post and this relationship

### PR DESCRIPTION
**Pull request by mvn-caonguyen-dn**
## Description
Delete post and this relationship
## Reference Requirements
- Product Backlog ID: PBI-11
- Task ID: #11.3

## Reference Isuees
#{issue_id}

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- resources/lang/en/posts.php 

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- delete from `likes` where `likes`.`post_id` = '2' and `likes`.`post_id` is not null
image
- update `comments` set `deleted_at` = '2018-01-29 10:41:27', `updated_at` = '2018-01-29 10:41:27' where `comments`.`commentable_id` = '2' and `comments`.`commentable_id` is not null and `comments`.`commentable_type` = 'post' and `comments`.`deleted_at` is null
image
- update `posts` set `deleted_at` = '2018-01-29 10:41:27', `updated_at` = '2018-01-29 10:41:27' where `id` = '2'
image

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
LGTM

File: resources/views/backend/posts/index.blade.php [hunk 1]
```diff
@@ -50,10 +51,10 @@
                   </tr>
                 </thead>
                 <tbody>
-                  @foreach ($posts as $post)
+                  @foreach ($posts as $index => $post)
                     <tr>
-                      <td class="text-center">{{ $post->id }}</td>
-                      <td><a href="{{ route('posts.show', $post->id) }}">{!! Str::words($post->content, config('define.posts.size_short_content'),config('define.posts.three_dots')) !!}</a></td>
+                      <td class="text-center">{{ $index + $posts->firstItem() }}</td>
+                      <td><a href="{{ route('posts.show', [$post->id, 'page' => $_GET['page'] ?? 1]) }}">{!! Str::words($post->content, config('define.posts.size_short_content'),config('define.posts.three_dots')) !!}</a></td>
```
Comment (at-nhuibo) [inline resources/views/backend/posts/index.blade.php]: Không được sử dụng `$_GET` vs `$_POST` thuần của PHP nha, vì nó ko an toàn và bảo mật kém.
Dùng `request('page')` của Laravel á.

File: resources/views/backend/posts/show.blade.php [hunk 2]
```diff
@@ -71,9 +72,13 @@
               <div style="padding-top: 10px;"><b>{{ __('posts.content') }}</b></div>
               <p>{!! $post->content !!}</p>
             </ul>
-            <a href="#" class="btn btn-danger btn-block btn-flat btn-delete-item" data-title="{{ __('common.confirm.title') }}" data-confirm="{{ __('common.confirm.delete_post') }}">
-              <b>{{ __('posts.delete') }}</b>
-            </a>
+            <form method="POST" action="{{ route('posts.destroy', [$post->id, 'page' => $_GET['page'] ?? 1]) }}">
```
Comment (at-nhuibo) [inline resources/views/backend/posts/show.blade.php]: `$_GET` này cũm vậy.
À, ở chỗ list ko có nút delete hả? vào detail mới delete đc thôi hay sao?

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment vs nha!

File: resources/views/backend/posts/show.blade.php (hunk 2 above)
Comment (at-caonguyen) [reply resources/views/backend/posts/show.blade.php]: Dạ trong list post không có nút edit với delete đó anh

**Review by ghost — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**